test/viewmodels/home_view_model_integration_test.dart

```diff
@@ -0,0 +1,191 @@
+        const GeoPoint(1, 0), // This is >> 0.1 km away from the (0,0)
+      );
+
+      final actualPosts = await container.read(postOverviewProvider.future);
+
+      expect(actualPosts, isEmpty);
+    });
+
+    test("Single near post returned correctly", () async {
+      // Add the post owner to the database
+      final owner = MockUserFirestore.generateUserFirestore(1)[0];
+      await userRepo.setUser(owner.uid, owner.data);
+
+      // Add the post to the database
+      final postData = MockPostFirestore.generatePostData(1).map((postData) {
+        return PostData(
+          ownerId: owner.uid, // Map to the owner
+          title: postData.title,
+          description: postData.description,
+          publicationTime: postData.publicationTime,
+          voteScore: postData.voteScore,
+        );
+      }).first;
+
+      const postPosition =
+          GeoPoint(0.0001, 0); // This is < 0.1 km away from the (0,0)
+
+      await postRepo.addPost(
+        postData,
+        postPosition,
+      );
+
+      // Get the expected post overview
+      final expectedPosts = [
+        PostOverview(
+          title: postData.title,
+          description: postData.description,
+          votes: postData.voteScore,
+          posterUsername: owner.data.username,
+          commentNumber: 0,
+        ),
+      ];
+
+      final actualPosts = await container.read(postOverviewProvider.future);
+
+      expect(actualPosts, expectedPosts);
+    });
+
+    test("Throws an exception when the owner of a post is not found", () async {
+      // Add the post to the database
+      final postData = MockPostFirestore.generatePostData(1).first;
+
+      await postRepo.addPost(
+        postData,
+        userPosition,
+      );
+
+      expect(
+        container.read(postOverviewProvider.future),
+        throwsA(isA<Exception>()),
+      );
+    });
+
+    test("Multiple near posts with multiple owners returned correctly",
+        () async {
+      const nbOwners = 3;
+      const nbPosts = 10;
+
+      // Add the post owners to the database
+      final owners = MockUserFirestore.generateUserFirestore(nbOwners);
+      for (final owner in owners) {
+        await userRepo.setUser(owner.uid, owner.data);
+      }
+
+      // Add the posts to the database
+      final postDatas = MockPostFirestore.generatePostData(nbPosts)
+          .mapIndexed(
+            (index, element) => PostData(
+              ownerId: owners[index % nbOwners].uid, // Map to an owner
+              title: element.title,
+              description: element.description,
+              publicationTime: element.publicationTime,
+              voteScore: element.voteScore,
+            ),
+          )
+          .toList();
+
+      // The 6 first posts are under 100m away from the user and are the ones expected
+      const nbPostsInRange = 6;
+      final postPositions = List.generate(nbPosts, (i) {
+        return GeoPoint(0.0001 + i * 0.0001, 0.0001 + i * 0.0001);
+      });
```

Comment:
I think that we can make this part a bit more explicit for future readers.

```suggestion
      final postInRange = List.generate(nbPostsInRange, (i) {
        return GeoPoint(0.0001 + i * 0.0001, 0.0001 + i * 0.0001);
      });

      //Generate post positions that are not in the range.
      // The distance between [userPosition = GeoPoint(0, 0)] and a GetPoint at
      // latitude 0.0006 and longitude 0.0006 is about 0.11 km.
      final postsNotRange = List.generate(nbPosts - nbPostsInRange, (i) {
        i = i + nbPostsInRange;
        return GeoPoint(0.0001 + i * 0.0001, 0.0001 + i * 0.0001);
      });

      final postPositions = [...postInRange, ...postsNotRange];
```

Reply:
Great suggestion !
This will make the test easier to understand.
Applied in 02d13f36dd00ab7f7efc8a720da36de868c38ff7.